src/package-hoister.js

```diff
@@ -240,6 +247,7 @@ export default class PackageHoister {
     const parts = parentParts.concat(pkg.name);
     const key: string = this.implodeKey(parts);
     const info: HoistManifest = new HoistManifest(key, parts, pkg, loc, isDirectRequire, isRequired, isIncompatible);
+    info.previousPatterns = new Set(parent == null ? [] : parent.previousPatterns).add(pattern);
```

Comment:
Set constructor works fine if the argument is undefined. It will create an empty Set, so we can skip this validation.
```suggestion
    info.previousPatterns = new Set(parent?.previousPatterns).add(pattern);
```